tortoise/recover.go

```diff
@@ -20,10 +21,11 @@ import (
 func Recover(
 	ctx context.Context,
 	db sql.Executor,
+	atxdata *atxsdata.Data,
 	current types.LayerID,
 	opts ...Opt,
 ) (*Tortoise, error) {
-	trtl, err := New(opts...)
+	trtl, err := New(atxdata, opts...)
```

Comment:
in recovery it would be the best if we iterate over atxs in atxdata, this will save a lot of time on db access